tests/common/test_cgroupstelemetry.py

```diff
@@ -162,3 +160,81 @@ def test_telemetry_calculations(self, *args):
                                  collected_metrics[name]["cpu"]["cur_cpu"][0:5])
 
         self.cleanup_cgroup_telemetry()
+
+    def test_cgroup_tracking(self):
+        self.cleanup_cgroup_telemetry()
+
+        num_extensions = 5
+        num_controllers = 2
+        for i in range(num_extensions):
+            dummy_cpu_cgroup = CGroup.create("dummy_path", "cpu", "dummy_extension_{0}".format(i))
+            CGroupsTelemetry.track_cgroup(dummy_cpu_cgroup)
+
+            dummy_memory_cgroup = CGroup.create("dummy_path", "memory", "dummy_extension_{0}".format(i))
+            CGroupsTelemetry.track_cgroup(dummy_memory_cgroup)
+
+        self.assertEqual(num_extensions * num_controllers, len(CGroupsTelemetry._tracked))
```

Comment:
a better assertion could be that is_tracked returns true for all the items you added (that also has the advantage of testing via the public interface, rather than reaching into _tracked)